jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/BuildImageStep.java

```diff
@@ -173,12 +174,43 @@ public DescriptorDigest getDiffId() throws LayerPropertyNotFoundException {
                 .build());
       }
       if (containerConfiguration != null) {
+        boolean shouldInheritEntrypoint = containerConfiguration.getEntrypoint() == null;
+        boolean shouldInheritProgramArguments =
```

Comment:
What do you think about breaking this out for readability.
```
imageBuilder.setEntryPoint(calculateEntrypoint(containerConfiguration, baseImage))
imageBuilder.setJavaArguments(calculateProgramArguments(containerConfiguration, baseImage))
```
and implement the `calculateX methods`

Reply:
we need these boolean for these logs : 
```
if (shouldInheritEntrypoint && baseImage.getEntrypoint() != null) {
          buildConfiguration
              .getEventDispatcher()
              .dispatch(
                  LogEvent.lifecycle(
                      "Container entrypoint set to "
                          + baseImage.getEntrypoint()
                          + " (inherited from base image)"));
        }
        if (shouldInheritProgramArguments && baseImage.getJavaArguments() != null) {
          buildConfiguration
              .getEventDispatcher()
              .dispatch(
                  LogEvent.lifecycle(
                      "Container program arguments set to "
                          + baseImage.getJavaArguments()
                          + " (inherited from base image)"));
        }
```